platform/security/expansion/security-expansion-impl/src/main/java/ddf/security/expansion/impl/AbstractExpansion.java

```diff
@@ -453,30 +450,49 @@ protected void loadConfiguration(String filename) {
             // absolute path
             file = new File(filename);
         }
-        try (BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(file),
-                StandardCharsets.UTF_8.name()))) {
-            if (br != null) {
-                String line;
-                while ((line = br.readLine()) != null) {
-                    if ((line.length() > 0) && (!line.startsWith(CFG_COMMENT_STR))) {
-                        if (line.startsWith(SEPARATOR_PREFIX)) {
-                            if (line.length() > SEPARATOR_PREFIX.length()) {
-                                attributeSeparator = line.substring(SEPARATOR_PREFIX.length());
-                            } else {
-                                attributeSeparator = DEFAULT_VALUE_SEPARATOR;
-                            }
-                        } else {
-                            addExpansionRule(line);
-                        }
-                    }
-                }
-                LOGGER.info("Finished loading mapping configuration file.");
-            }
+
+        try (InputStream is = new FileInputStream(file)){
+            loadConfiguration(is);
         } catch (IOException e) {
             LOGGER.warn("Unexpected exception reading mapping configuration file {} - {}",
                     filename,
                     e.getMessage());
             setExpansionMap(null);
         }
     }
+
+    /**
+     * Loads an expansion configuration from an inputstream.
+     * It is the responsiblity of the caller to close the stream.
+     *
+     * @param is an inputstream to the resource containing the expansion configuration
+     * @throws IOException if there is an error processing the stream
+     */
+    void loadConfiguration(InputStream is) throws IOException {
+        // first clear out the existing table
+        if (expansionTable != null) {
+            expansionTable.clear();
+        }
+
+        if (is == null) {
+            return;
+        }
+
+        BufferedReader br = new BufferedReader(new InputStreamReader(is,
+                StandardCharsets.UTF_8.name()));
+        br.lines()
+                .filter(line -> !line.isEmpty())
+                .filter(line -> !line.startsWith(CFG_COMMENT_STR))
+                .forEach(line -> {
+                    if (line.startsWith(SEPARATOR_PREFIX)) {
```

Comment:
Is it possible `.filter` on `line.startsWith(SEPARATOR_PREFIX)` and `line.length > SEPARATOR.PREFIX.length()?   Then you wouldn't need any `if` statements in the `forEach` because the lines will already be filtered.